packages/@glimmer/runtime/lib/compiled/opcodes/expressions.ts

```diff
@@ -105,6 +114,14 @@ APPEND_OPCODES.add(Op.JitSpreadBlock, vm => {
   }
 });
 
+function isUndefinedReference(input: JitScopeBlock | Reference): input is Reference {
+  assert(
+    Array.isArray(input) || input === UNDEFINED_REFERENCE,
+    'a reference other than UNDEFINED_REFERENCE is illegal here'
+  );
+  return input === UNDEFINED_REFERENCE;
+}
+
 APPEND_OPCODES.add(Op.HasBlock, vm => {
```

Comment:
Should be updated to use the same type of guard that was added just below to HasBlockParams.

Reply:
Done!